File: day_1/solution.py
with open("input_day_1.txt", "r") as data:

    list1 = []
    list2 = []
    distance = 0

    for line in data:
        numbers = line.strip().split()
        list1.append(int(numbers[0]))
        list2.append(int(numbers[1]))

    list1.sort()
    list2.sort()

    # Part 1
    for num1, num2 in zip(list1, list2):
        distance_each_number = abs(num1 - num2)
        distance += distance_each_number

    # Part 2
    similarity_score = 0
    for num in list1:
        similarity_score += num * list2.count(num)

print(distance)
print(similarity_score)


    

File: day_1/solution.cpp
#include <iostream>
#include <fstream>
#include <sstream>
#include <vector>
#include <algorithm>
#include <cmath>  // For abs()

using namespace std;

int main() {
    string line;
    ifstream dataFile("C:\\Users\\stran\\Documents\\AdventCode2024\\day_1\\input_day_1.txt");

    if (dataFile.is_open()) {
        vector<int> list1, list2;

        while (getline(dataFile, line)) {
            stringstream ss(line); // Parse the line
            int num1, num2;
            ss >> num1 >> num2;
            list1.push_back(num1);
            list2.push_back(num2);
        }
        dataFile.close();

        // Sorting both lists
        sort(list1.begin(), list1.end());
        sort(list2.begin(), list2.end());

        // Part 1: Calculate the total distance
        int distance = 0;
        for (size_t i = 0; i < list1.size(); ++i) {
            distance += abs(list1[i] - list2[i]);
        }

        // Part 2: Calculate the similarity score
        int similarity_score = 0;
        for (const int& num : list1) {
            similarity_score += num * count(list2.begin(), list2.end(), num);
        }

        // Output the results
        cout << "Total Distance: " << distance << endl;
        cout << "Similarity Score: " << similarity_score << endl;
    } else {
        cout << "Unable to open file" << endl;
    }

    return 0;
}
